.github/workflows/run_teaspoons_e2e_cli_tests.yaml

```diff
@@ -0,0 +1,181 @@
+          service_account: 'firecloud-qa@broad-dsde-qa.iam.gserviceaccount.com'
+          access_token_scopes: 'profile, email, openid'
+          access_token_subject: 'hermione.owner@quality.firecloud.org'
+          export_environment_variables: false
+      
+      - name: Generate OAuth2 2.0 access token for admin user and billing project owner
+        id: obtain_admin_user_token
+        uses: google-github-actions/auth@v1
+        with:
+          token_format: 'access_token'
+          workload_identity_provider: 'projects/1038484894585/locations/global/workloadIdentityPools/github-wi-pool/providers/github-wi-provider'
+          service_account: 'firecloud-qa@broad-dsde-qa.iam.gserviceaccount.com'
+          access_token_scopes: 'openid, https://www.googleapis.com/auth/userinfo.email, https://www.googleapis.com/auth/userinfo.profile, https://www.googleapis.com/auth/cloud-billing'
+          access_token_subject: 'dumbledore.admin@quality.firecloud.org'
+          export_environment_variables: false
+      
+      - name: Setup Teaspoons resources
+        id: setup-resources
+        env:
+          BEE_NAME: ${{ inputs.bee-name }}
+          BILLING_PROJECT_NAME: ${{ inputs.billing-project-name }}
+          BILLING_ACCOUNT_NAME: 'billingAccounts/00708C-45D19D-27AAFA'
+          ADMIN_TOKEN: ${{ steps.obtain_admin_user_token.outputs.access_token }}
+          WDL_METHOD_VERSION: ${{ inputs.wdl-method-version }}
+          WORKSPACE_NAME: 'teaspoons_e2e_cli_test_workspace'
+        run: |
+          cd dsp-reusable-workflows/
+          python e2e-test/teaspoons_gcp_e2e_cli_test_setup.py
+        
+      - name: Run e2e test with CLI
+        id: run-cli-e2e-test
+        env: 
+          BEE_NAME: ${{ inputs.bee-name }}
+          USER_TOKEN: ${{ steps.obtain_user_token.outputs.access_token }}
+          MAX_WAIT_TIME: 1800  # 30 min in seconds
+          CHECK_INTERVAL: 120  # 2 minutes in seconds
+          INITIAL_WAIT: 600    # 10 minutes in seconds
+        run: |
+          set -eu
+          
+          cd terra-scientific-pipelines-service-cli/
+          poetry install
+
+          echo "### CONFIGURING CLI ###"
+          CONFIG_FILE=$(poetry run python -c "import terralab; print(terralab.__file__.replace('__init__.py','.terralab-cli-config'))")
+          echo "found config file at $CONFIG_FILE"
+          [ -f "$CONFIG_FILE" ] || { echo "Config file not found"; exit 1; }
+          
+          TEASPOONS_API_URL="https://teaspoons.$BEE_NAME.bee.envs-terra.bio"
+          sed -i -e "s|^TEASPOONS_API_URL=.*|TEASPOONS_API_URL=$TEASPOONS_API_URL|" "$CONFIG_FILE"
+          grep TEASPOONS_API_URL "$CONFIG_FILE" || { echo "Failed to update config"; exit 1; }
+
+          echo "### RUNNING CLI TEST ###"
+          poetry run terralab --debug login-with-oauth "$USER_TOKEN"
+
+          touch input_file.vcf.gz
```

Comment:
super nitpick but this could be put into a variable so its consistent with the line that submits the job. it could also just be added to the `env` block for this step like `MAX_WAIT_TIME`